tool/tsh/tsh_test.go

```diff
@@ -231,11 +240,24 @@ func TestOIDCLogin(t *testing.T) {
 	// verify that auto-request happened
 	require.True(t, didAutoRequest.Load())
 
+	findMOTD(t, sc, motd)
 	// if we got this far, then tsh successfully registered name change from `alice` to
 	// `alice@example.com`, since the correct name needed to be used for the access
 	// request to be generated.
 }
 
+func findMOTD(t *testing.T, sc *bufio.Scanner, motd string) {
+	t.Helper()
+	found := false
+	for sc.Scan() {
+		if strings.Contains(sc.Text(), motd) {
+			found = true
+			break
+		}
+	}
+	require.True(t, found, "Failed to find %q MOTD in the logs", motd)
```

Comment:
nit: I don't think `found`variable is needed here. You could replace the break a few lines above with `return` and this line with `require.Fail`.  In my opinion that would simplify this function.